shanoir-ng-datasets/src/main/java/org/shanoir/ng/dataset/service/CreateStatisticsService.java

```diff
@@ -0,0 +1,120 @@
+import java.util.stream.Collectors;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+@Service
+public class CreateStatisticsService {
+    @Autowired
+    ShanoirEventService eventService;
+    @Autowired
+    private DatasetService datasetService;
+    private static final String ZIP = ".zip";
+    private static final String JAVA_IO_TMPDIR = "java.io.tmpdir";
+    private static final Logger LOG = LoggerFactory.getLogger(CreateStatisticsService.class);
+
+    private File recreateFile(final String fileName) throws IOException {
+        File file = new File(fileName);
+        if(file.exists()) {
+            file.delete();
+        }
+        file.createNewFile();
+        return file;
+    }
+
+    @Async
+    @Transactional
+    public void createStats(String studyNameInRegExp, String studyNameOutRegExp, String subjectNameInRegExp, String subjectNameOutRegExp, ShanoirEvent event, String params) throws RestServiceException, IOException {
+        float progress = 0;
+        String tmpDir = System.getProperty(JAVA_IO_TMPDIR);
+        File userDir = DatasetFileUtils.getUserImportDir(tmpDir);
+        File statisticsFile = recreateFile(userDir + File.separator + "shanoirExportStatistics.tsv");
+        File zipFile = recreateFile(userDir + File.separator + "shanoirExportStatistics_" + event.getId() + ZIP);
+
+        // Get the data
+        try (FileOutputStream fos = new FileOutputStream(statisticsFile);
+             BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(fos))){
+
+            List<Object[]> results = datasetService.queryStatistics(studyNameInRegExp, studyNameOutRegExp, subjectNameInRegExp, subjectNameOutRegExp);
+
+            for (Object[] or : results) {
+
+                progress += 1f / results.size();
+                event.setProgress(progress);
+                eventService.publishEvent(event);
+                List<String> strings = Arrays.stream(or).map(object -> Objects.toString(object, null)).collect(Collectors.toList());
+                bw.write(String.join("\t", strings));
+                bw.newLine();
+            }
+
+        } catch (jakarta.persistence.NoResultException e) {
+            event.setStatus(ShanoirEvent.ERROR);
+            event.setMessage("No statistics found.");
+            event.setProgress(-1f);
+            eventService.publishEvent(event);
+            throw new RestServiceException(new ErrorModel(HttpStatus.NOT_FOUND.value(), "No result found.", e));
+        } catch (Exception e) {
+            event.setStatus(ShanoirEvent.ERROR);
+            event.setMessage("Error during fetching of statistics.");
+            event.setProgress(-1f);
+            eventService.publishEvent(event);
+            throw new RestServiceException(
+                    new ErrorModel(HttpStatus.UNPROCESSABLE_ENTITY.value(), "Error while querying the database.", e));
+        }
+
+        zipSingleFile(statisticsFile, zipFile);
+        statisticsFile.delete();
```

Comment:
This could be in a "finally" clause at the end of the method to ensure that this file is deleted whatever the result.